### PR TITLE
12495: add documentation on observability local dev setup

### LINKED ISSUE #12495: Connect local dev environment Log4j logs to Demo Environment's App Insights
### User Story

As a ReportStream engineer implementing Application Insights improvements, I want to test configuration changes on my local dev environment so that I do not have to merge changes to master to see if my changes worked on staging.

### Description/Use Case

DevOps is working on enabling [distributed tracing](https://learn.microsoft.com/en-us/azure/azure-monitor/app/monitor-functions#distributed-tracing-for-java-applications ) for the apps running in azure. At the same time, platform team should look into how to set up distributed tracing locally (this ticket) so we can test subsequent tickets belonging to #12458 without having to merge to master.

### Dev Notes

> Note: These dev notes include the understandings coming out of the initial investigation required to create this ticket. All claims should be investigated and independently verified. An alternate route should be proposed if the acceptance criteria cannot be met by following the dev notes.

This ticket will have the two main parts:
1. Connect local dev environment to one of the demo application App Insights. 
2. Enable distributed tracing. 

Both of these seem to be accomplished by [running Microsoft's OpenTelemetry .jar as an agent](https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-enable?tabs=aspnetcore). This will probably require changes to the docker scripts, which run the application locally. See the documentation for [configuring a Java application for application insights](https://learn.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config).

This [overview document](https://learn.microsoft.com/en-us/azure/azure-monitor/app/monitor-functions) is a good start for getting up to speed on azure functions integration with application insights and verifying these dev notes (which you should do). ReportStream already has a basic integration with Application Insights and, as discussed in the aforementioned document, the java agent is needed to get the full benefits of the integration. Integrating the [OpenTelemetry](https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-overview?tabs=aspnetcore) jar will allow ReportStream to send custom metrics, events, and structured logs to Application Insights, greatly increasing the usability and helpfulness of it.

For testing or troubleshooting, contact Arnej for CURL command to send app insights data to staging env from command line.

### Acceptance Criteria

- [x] When running the application locally, all traces, exceptions, and logs (including log4j logs) go to a demo environment's Application Insights
- [x] Documentation created on how to configure local app to connect to demo environment app insights

### PR DESCRIPTION
**Pull request by jalbinson**
This PR adds documentation around connecting your local machine to appinsights for observability.

## Changes
- New doc

## Linked Issues
- Fixes #12495

### CODE REVIEW

File: prime-router/docs/observability/local-dev-setup.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+4. Run the `quickRun` Gradle task ensuring that the new environment variable is being picked up.
+   1. Make sure everything starts up normally 
+   2. Make a few calls to an authenticated endpoint
+   3. I suggested this one: http://localhost:7071/api/waters/report/1/history
+   4. Wait about 10 minutes and ensure data is flowing into the appinsights portal. You should be able to see basic statistics around the requests.
+    
+## Extra Resources
+
+- [Official Microsoft Documentation](https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-enable?tabs=java)
+
```
Comment (arnejduranovic) [inline prime-router/docs/observability/local-dev-setup.md:30]: Empty lines?

File: prime-router/docs/observability/local-dev-setup.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Connect Local Environment to appinsights
+
+## Prerequisites
+ - CDC account 
+ - SU access
```
Comment (arnejduranovic) [inline prime-router/docs/observability/local-dev-setup.md:5]: nit: I would combine this one with the below point and refactor to: `Azure Portal CDC Super User Access`

Comment (jalbinson) [reply prime-router/docs/observability/local-dev-setup.md:5]: will do

File: prime-router/docs/observability/local-dev-setup.md (hunk 1 above)
Comment (jalbinson) [reply prime-router/docs/observability/local-dev-setup.md:30]: wow thats a lot of empty lines 😄

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**